File: code/server/src/github.com/jason-wilmans/votesComponent/domainObjects/Option.go
package domainObjects

import (
	"github.com/satori/go.uuid"
	"github.com/muroc/prego"
	"github.com/jason-wilmans/infrastructure/goExtensions"
)

type Option struct {
	Id			string
	VoteId		string
	Name		string
	Description string
}

func NewOption(voteId string, name string, description string) *Option {
	precond.NotNil(voteId, "")
	precond.NotNil(name, "")
	precond.True(goExtensions.IsNonWhitespaceWithMinimumLength(name, 1), "")

	id, _ := uuid.NewV4()
	return &Option{id.String(), voteId,name,	description}
}

func (this *Option) GetId() uuid.UUID {
	id, _ := uuid.FromString(this.Id)
	return id
}

func (this *Option) SetName(name string) {
	this.Name = name
}

func (this *Option) GetName() string {
	return this.Name
}

func (this *Option) SetDescription(description string) {
	this.Description = description
}

func (this *Option) GetDescription() string {
	return this.Description
}

func (this *Option) GetVoteId() uuid.UUID {
	id, _ := uuid.FromString(this.VoteId)
	return id
}

File: code/webClient/src/app/model/option.ts
export class Option {
  private readonly Id: string;
  private readonly VoteId: string;
  private Name: string;
  private Description: string;

  constructor(voteId: string, name: string, description?: string, id?: string) {
    this.VoteId = voteId;
    this.Name = name;
    this.Description = description;
    if(id) {
      this.Id = id;
    }
  }

  public GetId() : string {
    return this.Id;
  }

  public GetVoteId(): string {
    return this.VoteId;
  }

  public SetName(name: string) {
    this.Name = name;
  }

  public GetName() : string {
    return this.Name;
  }

  public SetDescription(description: string) {
    this.Description = description;
  }

  public GetDescription() : string {
    return this.Description;
  }
}
